packages/perseus/src/components/tooltip.tsx

```diff
@@ -40,136 +41,135 @@
-            borderTop = vBorder;
-        } else {
-            borderBottom = vBorder;
-        }
-
-        return (
-            <div
-                style={{
-                    display: "block",
-                    height: 0,
-                    width: 0,
-                    position: "absolute",
-                    left: this.props.left,
-                    top: this.props["top"],
-                    borderLeft: borderLeft,
-                    borderRight: borderRight,
-                    borderTop: borderTop,
-                    borderBottom: borderBottom,
-                }}
-            />
-        );
-    },
-});
-
-const TooltipArrow = createReactClass({
-    propTypes: {
-        position: PropTypes.string,
-        visibility: PropTypes.string,
-        left: PropTypes.number,
-        top: PropTypes.number,
-        color: PropTypes.string.isRequired, // a css color
-        border: PropTypes.string.isRequired, // a css color
-        width: PropTypes.number.isRequired,
-        height: PropTypes.number.isRequired,
-        horizontalDirection: PropTypes.oneOf(["left", "right"]).isRequired,
-        verticalDirection: PropTypes.oneOf(["top", "bottom"]).isRequired,
-    },
+import type {Property} from "csstype";
 
-    getDefaultProps: function () {
-        return {
-            position: "relative",
-            visibility: "visible",
-            left: 0,
-            top: 0,
-        };
-    },
+const zIndex = 10;
 
+type HorizontalDirection = "left" | "right";
+type VerticalDirection = "top" | "bottom";
+
+type TriangleProps = {
+    color: string;
+    left: number;
+    top: number;
+    width: number;
+    height: number;
+    horizontalDirection: HorizontalDirection;
+    verticalDirection: VerticalDirection;
+};
+
+const Triangle = (props: TriangleProps) => {
+    let borderLeft;
+    let borderRight;
+    let borderTop;
+    let borderBottom;
+
+    const hBorder = `${props.width}px solid transparent`;
+    if (props.horizontalDirection === "right") {
+        borderLeft = hBorder;
+    } else {
+        borderRight = hBorder;
+    }
+
+    const vBorder = `${props.height}px solid ${props.color}`;
+    if (props.verticalDirection === "top") {
+        borderTop = vBorder;
+    } else {
+        borderBottom = vBorder;
+    }
+
+    return (
+        <div
+            style={{
+                display: "block",
+                height: 0,
+                width: 0,
+                position: "absolute",
+                left: props.left,
+                top: props["top"],
+                borderLeft: borderLeft,
+                borderRight: borderRight,
+                borderTop: borderTop,
+                borderBottom: borderBottom,
+            }}
+        />
+    );
+};
+
+type TooltipArrowProps = {
+    position: Property.Position;
+    visibility: Property.Visibility;
+    left: number;
+    top: number;
+    color: string; // a css color
+    border: string; // a css color
```

Comment:
This is good enough for now, but we could define a type for CSS colors in the future.  Something like:
```
type RGB = `rgb(${number}, ${number}, ${number})`;
type RGBA = `rgba(${number}, ${number}, ${number}, ${number})`;
type HEX = `#${string}`;

type Color = RGB | RGBA | HEX;
```
See https://stackoverflow.com/questions/42584228/how-can-i-define-a-type-for-a-css-color-in-typescript.

Reply:
Do you think it'd make sense to bake that into WS or WB?

Reply:
Maybe.  Aphrodite is another possible home for a `Color` type.  TBH, I'm not sure how useful having such a type is.  I might experiment with it when I have the time.